2022/incoming/santas-helper-embrace-the-shell.pod

```diff
@@ -0,0 +1,141 @@
+Author: Stefan Adams <stefan@adams.fm>
+Title: Santa's Helper Embrace the Shell
+Topic: Command Line
+
+=encoding utf8
+
+=head1 Santa's Helpers Embrace the Shell
+
+Santa's workshop is very busy and everyone on staff is expected to contribute beyond just their traditional roles. Even Santa's reindeer take on additional tasking throughout the year.
+
+Santa and the elves have been busy this year making vast improvements to their processes, incorporating modern technology such as Perl into their daily workflows.
+
+To improve their real-time efficiency when it comes to irregular activity, debugging, and general proof-of-concept research, everyone has been working to improve their command line fu. Much of what they do involves working with structured data, such as JSON files.
+
+=head2 Typical reindeer...
+
+The reindeer have been helping by extracting important data from the elves' structured notes stored in JSON format. The reindeer need to get the GPS coordinates for the next deliveries in the route.
+
+    {"children":[{"list":"naughty","name":"Tom","position":[40.7414728,-74.0055813]},{"location":[40.7565323,-73.9904037],"name":"Dick","list":"nice"},{"name":"Harry","list":"nice","location":[40.7411595,-74.0097167]}]}
+
+The reindeer use this data to determine the next deliveries on Santa's route:
+
+    $ perl -nE 'while (s/"name":"([^"]+)","list":"nice","(position|location)":\[([^,]+),([^\]]+)//) { say "$1\t$3\t$4" }' < route.json
+    Harry   40.7411595      -74.0097167
+
+The elves confirmed that Tom, who is naughty, did not get added to Santa's route, but they also noticed that Dick, who is nice, is missing from the list! The reindeers with their poor JSON parser almost caused Dick to get missed! D:
+
+The elves knew this was not a sustainable approach. They decided that they must adopt better tooling to avoid any tragedies with incorrectly skipping a delivery for a child on Christmas day.
+
+=head2 The elves find C<jp>
+
+The elves discovered a new command line tool for parsing these JSON files and it allows the use of pointers to drill down to the specific data of interest.  This new tool is called C<jp> -- for JSON Pointer.
+
+The key features for C<jp> -- made possible by L<Mojo::JSON|https://docs.mojolicious.org/Mojo/JSON> from the wonderful L<Mojolicious|https://mojolicious.org> framework -- are that the elves can use a simple JSON pointer to get at any specific data in their JSON data structure, but then also use Perl data structures to apply logic for ensuring that they extract exactly the right data and not accidentally route Santa to someone on his naughty list.
```

Comment:
"to get at any specific data in their JSON" - I guess this means any specific data set in which case perhaps just explicitly say "data set" here.